From 1a5686508c8099c4e7929727fcbfd517abea2f67 Mon Sep 17 00:00:00 2001
From: Yifan Hong <elsk@google.com>
Date: Tue, 14 Aug 2018 10:55:14 -0700
Subject: [PATCH] System always contains root dir.

In recovery, system image is mounted to /mnt/system. After the work of
first stage ramdisk, system image always contains the root dir, no
matter what value of ro.build.system_root_image is.

Test: do compat check in recovery
Bug: 112586588
Change-Id: I3890c9c1335b23fd22a8910bca07e59d631bac71
---
 VintfObjectRecovery.cpp | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)

diff --git a/VintfObjectRecovery.cpp b/VintfObjectRecovery.cpp
index 9d7cab7..fb2da0f 100644
--- a/VintfObjectRecovery.cpp
+++ b/VintfObjectRecovery.cpp
@@ -47,11 +47,7 @@ class RecoveryPartitionMounter : public PartitionMounter {
         if (fstab == NULL) {
             return UNKNOWN_ERROR;
         }
-        if (getPropertyFetcher().getBoolProperty("ro.build.system_root_image", false)) {
-            return mountAt(fstab, "/", "/system_root");
-        } else {
-            return mountAt(fstab, "/system", "/system");
-        }
+        return mountAt(fstab, "/", "/system_root");
     }
 
     status_t mountVendor() const override {
@@ -63,11 +59,7 @@ class RecoveryPartitionMounter : public PartitionMounter {
     }
 
     status_t umountSystem() const override {
-        if (getPropertyFetcher().getBoolProperty("ro.build.system_root_image", false)) {
-            return umount("/system_root");
-        } else {
-            return umount("/system");
-        }
+        return umount("/system_root");
     }
 
     status_t umountVendor() const override {
-- 
2.17.1

